pkg/controller/service/servicecontroller.go

```diff
@@ -422,10 +422,15 @@ func (s *serviceCache) delete(serviceName string) {
 }
 
 func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api.Service) bool {
-	if !wantsLoadBalancer(oldService) && !wantsLoadBalancer(newService) {
+	oldServiceNeedsLoadBalancer := wantsLoadBalancer(oldService)
```

Comment:
Just to clarify I'm not against doing just, just conveying my caution so you verify the worst case doesn't happen